// @moralisweb3/next/lib/hooks/resolvers/useResolverPaginated.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = require("react");
var swr_1 = __importDefault(require("swr"));
var utils_1 = require("../../utils");
var useResolverPaginated = function (_a) {
    var endpoint = _a.endpoint, operation = _a.operation, request = _a.request, fetchParams = _a.fetchParams;
    var _b = (0, swr_1.default)([endpoint, { operation: operation, request: request }], request ? utils_1.fetcherPaginated : null, __assign({ revalidateOnFocus: false }, fetchParams)), data = _b.data, error = _b.error, mutate = _b.mutate, isValidating = _b.isValidating;
    var fetch = (0, react_1.useCallback)(function (params) {
        var fetchRequest = params !== null && params !== void 0 ? params : request;
        if (!fetchRequest) {
            throw new utils_1.NoHookParamsError(operation.name);
        }
        return mutate((0, utils_1.fetcherPaginated)(endpoint, {
            operation: operation,
            request: fetchRequest,
        }));
    }, []);
    return {
        data: data,
        error: error,
        fetch: fetch,
        isFetching: isValidating,
    };
};
exports.default = useResolverPaginated;
