// @moralisweb3/next/lib/hooks/evmApi/generated/token/useEvmTokenMetadataBySymbol.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useEvmTokenMetadataBySymbol = void 0;
var common_evm_utils_1 = require("moralis/common-evm-utils");
var resolvers_1 = require("../../../resolvers");
var useEvmTokenMetadataBySymbol = function (request, fetchParams) {
    var _a = (0, resolvers_1.useResolver)({
        endpoint: 'evmApi/getTokenMetadataBySymbol',
        operation: common_evm_utils_1.getTokenMetadataBySymbolOperation,
        request: request,
        fetchParams: fetchParams,
    }), data = _a.data, error = _a.error, fetch = _a.fetch, isFetching = _a.isFetching;
    return {
        data: data,
        error: error,
        fetch: fetch,
        /**
         * @deprecated use `fetch()` instead
         */
        refetch: function () { return fetch(); },
        isFetching: isFetching,
        /**
         * @deprecated use `isFetching` instead
         */
        isValidating: isFetching,
    };
};
exports.useEvmTokenMetadataBySymbol = useEvmTokenMetadataBySymbol;
