// @moralisweb3/next/lib/hooks/auth/useAuthRequestChallengeEvm.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useAuthRequestChallengeEvm = void 0;
var common_auth_utils_1 = require("@moralisweb3/common-auth-utils");
var resolvers_1 = require("../resolvers");
var useAuthRequestChallengeEvm = function (request, fetchParams) {
    var _a = (0, resolvers_1.useResolver)({
        endpoint: 'auth/requestChallengeEvm',
        operation: common_auth_utils_1.requestChallengeEvmOperation,
        request: request,
        fetchParams: fetchParams,
    }), data = _a.data, error = _a.error, fetch = _a.fetch, isFetching = _a.isFetching;
    return {
        challenge: data,
        error: error,
        requestChallengeAsync: fetch,
        /**
         * @deprecated use `fetch()` instead
         */
        refetch: function () { return fetch(); },
        isFetching: isFetching,
        /**
         * @deprecated use `isFetching` instead
         */
        isValidating: isFetching,
    };
};
exports.useAuthRequestChallengeEvm = useAuthRequestChallengeEvm;
